src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,63 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter FORMAT = DateTimeFormatter.ofPattern("d.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        if (suitablePeriod(dateFrom, dateTo) == false) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        StringBuffer result = new StringBuffer("Отчёт за период " + dateFrom + " - " + dateTo);
+        int salary = 0;
+        for (String retVal: names) {
+            salary = 0;
+            for (int i = 0; i < data.length; i++) {
+                String [] oneString = data[i].split(" ");
+                if (oneString[1].equals(retVal) == true
+                        && suitableDate(dateFrom, dateTo, oneString[0]) == true) {
+                    salary += Integer.parseInt(oneString[2]) * Integer.parseInt(oneString[3]);
+                }
+            }
+            result.append("\n").append(retVal).append(" - ").append(salary);
+        }
+
+        return result.toString();
     }
+
+    public static LocalDate dateConvInData(String data) {
+        LocalDate docDate = LocalDate.parse(data, FORMAT);
+        return docDate;
+    }
+
+    public static boolean suitablePeriod(String firstDate, String secondDate)
+            throws ParseException {
+        if (dateConvInData(firstDate).compareTo(dateConvInData(secondDate)) > 0) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public static int afterBeforeEqualsDays(String firstDate, String secondDate)
+            throws ParseException {
+        if (dateConvInData(firstDate).isAfter(dateConvInData(secondDate)) == true) {
+            return 1;
+        } else if (dateConvInData(firstDate).isEqual(dateConvInData(secondDate)) == true) {
+            return 0;
+        } else {
+            return -1;
+        }
+    }
+
+    public static boolean suitableDate(String firstDate, String secondDate, String exactDate)
+            throws ParseException {
+        if (afterBeforeEqualsDays(firstDate, exactDate) <= 0
+                && afterBeforeEqualsDays(secondDate, exactDate) >= 0) {
+            return true;
```

Comment:
@IrinaDyakonova you can much more simplify such methods. You can just return the result of computation without the if-else statement. Do you agree with me?

Reply:
@boroda4436 i agree with you